src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -1071,6 +1072,15 @@ def create_python_udf_or_sp(
         statement_params=statement_params,
     )
 
+    if comment is not None:
+        object_signature_sql = f"{object_name}({','.join(input_sql_types)})"
+        comment_query = f"COMMENT ON {object_type.value.split('_')[-1]} {object_signature_sql} IS '{comment}'"
```

Comment:
What happens if comment has a single quote in it?

Reply:
Good catch. It would have failed. I added escaped single quotes in the comment and added more test cases.